src/alloc.c

```diff
@@ -396,6 +396,21 @@ jl_tuple_t *jl_tuple_fill(size_t n, jl_value_t *v)
     return tup;
 }
 
+DLLEXPORT jl_bytevec_struct_t jl_bytevec(const uint8_t *data, size_t n)
+{
+    jl_bytevec_struct_t b;
+    if (n < 2*sizeof(void*)) {
+        memcpy(b.here.data, data, n);
```

Comment:
If `n == 2*sizeof(void*) - 1`, then it looks like the string data cannot be NUL-terminated.  Julia currently guarantees NUL-termination (in [array.c](https://github.com/JuliaLang/julia/blob/cf99eba0ce4ca6f4e210133423a1f47512320321/src/array.c#L60)) so that strings can be passed to external C code expecting NUL-terminated strings.  

Or is the plan to implement this on top of `ByteVec`s, similar to how it is implemented for UTF-16 and UTF-32 strings (i.e. `str.data` is actually a bytevec of length 1 more than the length of the string, and contains an explicit NUL terminator).   This might be cleaner, although it is a subtle breaking change for any code that currently looks at the `data` field of strings.